Index: component/portal/src/main/java/org/exoplatform/portal/config/UserACL.java
===================================================================
--- component/portal/src/main/java/org/exoplatform/portal/config/UserACL.java	(revision 48044)
+++ component/portal/src/main/java/org/exoplatform/portal/config/UserACL.java	(working copy)
@@ -147,7 +147,7 @@
 
   public boolean hasPermission(Page page, String remoteUser) throws Exception {
     if(PortalConfig.USER_TYPE.equals(page.getOwnerType())){
-      if( remoteUser.equals(page.getOwnerId())){
+      if(page.getOwnerId().equals(remoteUser)){
         page.setModifiable(true);
         return true;
       } 
