Index: web/eXoResources/src/main/webapp/javascript/eXo/portal/UIPortalControl.js
===================================================================
--- web/eXoResources/src/main/webapp/javascript/eXo/portal/UIPortalControl.js	(revision 61861)
+++ web/eXoResources/src/main/webapp/javascript/eXo/portal/UIPortalControl.js	(working copy)
@@ -1,6 +1,5 @@
 function UIPortalControl() {
   this.scrollManagers = new Array();
-  this.t = 0;
 };
 /**
  * Change state of window
@@ -82,18 +81,20 @@
 	if(e.keyCode) code = e.keyCode;
 	else if (e.which) code = e.which;
 	if(code ==13) {
-		if(this.t != 13) {
-			var uiPortalLoginFormAction = document.getElementById("UIPortalLoginFormAction");
-			if(uiPortalLoginFormAction) {
-				uiPortalLoginFormAction.onclick() ;
-			}
-			else
-			{
-				if(executeScript)
-					eval(executeScript);
-			}
+		if (window.event) {
+			e.returnValue = false;
+		} else {
+			e.preventDefault();
 		}
-	  this.t = code;
+		var uiPortalLoginFormAction = document.getElementById("UIPortalLoginFormAction");
+		if(uiPortalLoginFormAction) {
+			uiPortalLoginFormAction.onclick() ;
+		}
+		else
+		{
+			if(executeScript)
+				eval(executeScript);
+		}
 	}
 };
 
