Index: web/portal/src/main/webapp/groovy/portal/webui/workspace/UIExoStart.gtmpl
===================================================================
--- web/portal/src/main/webapp/groovy/portal/webui/workspace/UIExoStart.gtmpl	(révision 67652)
+++ web/portal/src/main/webapp/groovy/portal/webui/workspace/UIExoStart.gtmpl	(copie de travail)
@@ -304,7 +304,7 @@
 */  
 %>
 <div class="UIExoStart" id="UIExoStart">
-  <div class="StartBarContainer ClearFix">
+  <div class="StartBarContainer">
     <div class="ExoStartButton ButtonNormal" onclick="$onclick" title="<%=_ctx.appRes("UIExoStart.tooltip.begin")%>"><span></span></div>
     <div class="VLineSeparator"><span></span></div>
     <div class="DigitalClockContainer">
@@ -332,7 +332,7 @@
       <div class="StartMenuML">
         <div class="StartMenuMR">
           <div class="StartMenuBG">      
-            <div class="MenuItemContainer ClearFix" style="top: -1000px;">
+            <div class="MenuItemContainer" style="top: -1000px;">
               <%
                 for(menu in uicomponent.getMenus())  {
                   for(item in menu)  {
