Index: web/eXoResources/src/main/webapp/javascript/eXo/webui/UIRightClickPopupMenu.js
===================================================================
--- web/eXoResources/src/main/webapp/javascript/eXo/webui/UIRightClickPopupMenu.js	(revision 61870)
+++ web/eXoResources/src/main/webapp/javascript/eXo/webui/UIRightClickPopupMenu.js	(working copy)
@@ -148,7 +148,8 @@
 		ctxMenuContainer.style.width = "auto" ;
 		ctxMenuContainer.style.width = ctxMenuContainer.offsetWidth + 2 + "px" ;
 		//end fix width
-		contextMenu.style.left = intLeft + "px";
+		//need to add 1 more pixel because IE8 will dispatch onmouseout event to contextMenu.parent
+		contextMenu.style.left = (intLeft + 1) + "px";
 	}	else {
 		//move context menu to center of screen to fix width
 		contextMenu.style.right = eXo.core.Browser.getBrowserWidth() * 0.5 + "px" ;
@@ -158,6 +159,10 @@
 		contextMenu.style.right = intLeft + "px" ;
 	}
 	ctxMenuContainer.style.width = ctxMenuContainer.offsetWidth + "px" ;
+	//need to add 1 more pixel because IE8 will dispatch onmouseout event to contextMenu.parent
+	if((eXo.core.Mouse.mouseyInClient + ctxMenuContainer.offsetHeight) <= eXo.core.Browser.getBrowserHeight()) {
+		intTop += 1
+	}
 	contextMenu.style.top = intTop + "px";
 };
 
