Index: web/eXoResources/src/main/webapp/javascript/eXo/portal/PortalDragDrop.js
===================================================================
--- web/eXoResources/src/main/webapp/javascript/eXo/portal/PortalDragDrop.js	(revision 48114)
+++ web/eXoResources/src/main/webapp/javascript/eXo/portal/PortalDragDrop.js	(working copy)
@@ -14,6 +14,9 @@
 
 PortalDragDrop.prototype.init = function(e) {
 	if (!e) e = window.event;
+	
+	if(e && e.preventDefault) e.preventDefault();
+	
 	if(((e.which) && (e.which == 2 || e.which == 3)) || ((e.button) && (e.button == 2)))	return;
 	
 	var DOMUtil = eXo.core.DOMUtil ;
