Index: webui/portal/src/main/java/org/exoplatform/portal/application/PortalStateManager.java
===================================================================
--- webui/portal/src/main/java/org/exoplatform/portal/application/PortalStateManager.java	(revision 64764)
+++ webui/portal/src/main/java/org/exoplatform/portal/application/PortalStateManager.java	(working copy)
@@ -93,7 +93,10 @@
     if (state != null) {
       if ((!(Safe.equals(pcontext.getRemoteUser(), state.getUserName())))
           || (!pcontext.getPortalOwner().equals(state.getUIPortalApplication().getOwner()))) {
-        clearSession(pcontext.getRequest().getSession());
+        
+        if (state.getUserName() != null) {
+          clearSession(pcontext.getRequest().getSession());
+        }
         state = null;
       }
     }
